File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
from functools import cmp_to_key
def solution(numbers):
    answer = ''
    
    s_list = [str(i) for i in numbers]
    
    # print(s_list)
    
    def cmp_ab(a,b):
        return int(a+b) - int(b+a)
    
    s_list = sorted(s_list , key=cmp_to_key(cmp_ab), reverse=True)
    
    # print(s_list)
    
    answer = ''.join(s_list)
    
    return str(int(answer))

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;
class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        
        List<Integer> numString = new ArrayList<>();
        for(int i=0; i< numbers.length; i++) {
            numString.add(numbers[i]);
        }
        // System.out.println(numString);
        Collections.sort(numString, (a,b) -> {
            String as = String.valueOf(a);
            String bs = String.valueOf(b);
            
            return -Integer.compare(Integer.parseInt(as+bs),Integer.parseInt(bs+as));
        });
        // System.out.println(numString);
        
        StringBuilder sb = new StringBuilder();
        for(int i=0; i<numString.size(); i++) {
            sb.append(numString.get(i));
        }
        
        if (sb.charAt(0) == '0') {
            return "0";
        }

        
        return sb.toString();
    }
}